src/main/java/org/broadinstitute/hellbender/engine/filters/OverclippedReadFilter.java

```diff
@@ -63,6 +90,92 @@ public boolean test(final GATKRead read) {
             prevOperator = element.getOperator();
         }
 
-        return(alignedLength >= minimumSequenceLength || softClipBlocks < minSoftClipBlocks);
+        return (alignedLength >= minAlignedBases || softClipBlocks < minSoftClipBlocks);
+    }
+
+    private boolean testMinSoftClippedRatio(final GATKRead read) {
+        int totalLength = 0;
+        int alignedLength = 0;
+        int numSoftClippedBases = 0;
+        for ( final CigarElement element : read.getCigarElements() ) {
+            if (element.getOperator() == CigarOperator.S) {
+                numSoftClippedBases += element.getLength();
+            } else if ( element.getOperator().consumesReadBases() ) {   // M, I, X, and EQ (S was already accounted for above)
+                alignedLength += element.getLength();
+            }
+
+            totalLength += element.getLength();
+        }
+
+        final double softClipRatio = ((double)numSoftClippedBases / (double)totalLength);
+
+        logger.debug( "  Num Soft Clipped Bases: " + numSoftClippedBases );
+        logger.debug( "  Read Length: " + read.getLength() );
+        logger.debug( "  Read Filter status: (" + softClipRatio + " > " + minimumSoftClippedRatio + ")" );
+        logger.debug( "  Read Filter status: " + (softClipRatio > minimumSoftClippedRatio) );
+        logger.debug( "  Aligned Length: " + alignedLength );
+
+        return (alignedLength >= minAlignedBases) && (softClipRatio > minimumSoftClippedRatio);
```

Comment:
This should either be || or you should drop the first clause

Reply:
OBE